src/Tribe/PUE/Package_Handler.php

```diff
@@ -70,9 +70,9 @@ public function filter_upgrader_pre_download( $reply, $package, WP_Upgrader $upg
 	 * @return bool
 	 */
 	protected function is_mt_package( $package ) {
-		if ( empty( $package )
-		     || ! preg_match( '!^(http|https|ftp)://!i', $package )
-		     || file_exists( $package )
```

Comment:
Well, we've got three tests here and because of lazy evaluation we're only going to reach the `file_exists()` test if `$package` is a non-empty string that looks like a URL (because if it _doesn't_ look like a URL, we'll already have bailed).

But do we really want to perform a stat on a remote URL? I don't think we do, I think the intention was to check if `$package` exists in the local filesystem (yet we already know it is a remote URL by this point, else we'd have bailed out already).

Also ... my understanding - and local testing bears this out - is that stat-like operations aren't supported by the HTTP(S) stream wrappers, so if we _do_ want to make an HTTP request and determine if `$package` returns a 200 OK we'd want something more suited to that purpose in here, I'd have thought.

If you've still got reservations though I'm happy to rework: we could capture the match from the regex then see if that is in the list of returned stream wrappers _prior_ to calling `file_exists()`.